appIcons.js

```diff
@@ -97,7 +99,11 @@ const MyAppIcon = new Lang.Class({
                    'custom-theme-running-dots-border-width'];
 
         keys.forEach(function(key) {
-            this._dtdSettings.connect('changed::' + key, Lang.bind(this, this._toggleDots));
+            this._signalsHandler.add([
+                this._dtdSettings,
+                'changed::' + key,
+                Lang.bind(this, this._toggleDots)
+            ]);
```

Comment:
**Settings per dock**

I hadn't thought about it. I can check if it's simpler or not, I can't really tell quickly. I'll get back to you on this point.

**Completely reloading the extension**
When toggling multi-monitor, when a monitor changes or when the position is changed, I'm removing all docks and creating new ones within the dockManager class. In particular I found this helps a bit with the feeling of 'jerkiness' when toggling the multi-monitor option, or connecting a new monitor.

For instance, when disabling/enabling the whole extension we briefly reintroduce the stock dash, and then remove it again.